core/jvm/src/main/scala/cats/effect/unsafe/FiberMonitor.scala

```diff
@@ -51,6 +51,9 @@ private[effect] final class FiberMonitor(
   private[this] val size: Int = Runtime.getRuntime().availableProcessors() << 2
   private[this] val bags: Array[Map[AnyRef, WeakReference[IOFiber[_]]]] =
     new Array(size)
+  private[this] val extraBags: Set[Map[AnyRef, WeakReference[IOFiber[_]]]] =
+    // A synchronized weak set
+    Collections.newSetFromMap(Collections.synchronizedMap(new WeakHashMap()))
```

Comment:
Actually the order should probably be reversed.

```suggestion
    Collections.synchronizedSet(Collections.newSetFromMap(new WeakHashMap()))
```